DHFPROD-1453: hubSaveIndexes gradle task to save indexes

rahulvudutala opened the pull request:
This PR has commits regarding

* gradle task for saving indexes from entities on file system
* Changes in saveDbIndexes tests in marklogic-data-hub
* ml-data-hub unit test for hubSaveIndexes task

grechaw commented on marklogic-data-hub/build.gradle:
```diff
@@ -32,6 +32,7 @@ dependencies {
     compile 'com.marklogic:marklogic-data-movement-components:1.0'
     compile 'commons-io:commons-io:2.4'
     compile 'org.apache.commons:commons-text:1.1'
+    compile 'org.json:json:20180813'
```
you should not need to include this.  I'll find where you imported it and suggest an alternative

grechaw commented on marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java:
```diff
@@ -128,15 +136,23 @@ private boolean deployQueryOptions(DatabaseClient client, String filename) {
-            long lastModified = Math.max(finalFile.lastModified(), stagingFile.lastModified());
-            List<JsonNode> entities = getModifiedRawEntities(lastModified);
+            List<JsonNode> entities = getAllEntities();
+            List<String> databaseEntities = getEntityNamesInDatabase();
```
You can remove this part and accompanying function.  just stick with all entities.

grechaw commented on marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java:
```diff
@@ -193,6 +207,31 @@ private HubModuleManager getPropsMgr() {
         return entities;
     }
 
+    private List<String> getEntityNamesInDatabase() {
```
You'll want to remove this too -- we don't want to check database state to decide what to do with the filesystem

grechaw commented on ml-data-hub-plugin/src/test/groovy/com/marklogic/gradle/task/BaseTest.groovy:
```diff
@@ -144,6 +148,19 @@ class BaseTest extends Specification {
+		return BaseTest.class.getClassLoader().getResourceAsStream(resourceName)
+	}
+	
+	protected JSONObject getJsonResource(String fileLoc) {
```
use jackson's API rather tha JSONObject here.  The type is JsonNode and we already use it everywhere.

rahulvudutala commented:
I have addressed the comments. But I have amended few changes and did a force push to the previous commit. The changes in the previous commit are not reflected here.

grechaw commented:
@rahulvudutala can you please rebase to develop and re-submit.  This PR is a little messed up and I don't think the work is complete.

aebadirad commented:
See #1477 for the cleaned up branch
